File: backend/app/models/brand.py
"""Brand data models."""
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class BrandCreate(BaseModel):
    """Request model for creating a brand profile."""
    name: str
    description: Optional[str] = None
    keywords: list[str] = []
    domains: list[str] = []


class Brand(BrandCreate):
    """Brand profile stored in the system."""
    id: str
    created_at: datetime
    updated_at: datetime
    reputation_score: Optional[float] = None
    active_scouts: int = 0
    total_mentions: int = 0

File: frontend/src/types/brand.ts
/** Brand profile */
export interface Brand {
  id: string;
  name: string;
  description?: string;
  keywords: string[];
  domains: string[];
  created_at: string;
  updated_at: string;
  reputation_score?: number;
  active_scouts: number;
  total_mentions: number;
}

export interface BrandCreate {
  name: string;
  description?: string;
  keywords: string[];
  domains: string[];
}
